### DISCUSSION TITLE
Can the functions be modified to work from other data sourcers?

### DISCUSSION THREAD

**Discussion by rogerio-bio** [r-a-dobson/dynamicSDM#5] [Ideas]
Hello,

Firstly, congratulations on the package.

Is it possible to adapt the functions to work with other variable sources? For example, the use of CHELSA climate data? Using GEE in R is very complicated, and many errors appear. The program spends several minutes trying to solve a problem with the packages.

**Comment by r-a-dobson** [comment]
Hi Rogério - I will have a look into whether this is possible to set up, and keep this discussion updated!

**Reply by r-a-dobson** [reply]
An update on this:

Only global rasters of daily climate data can be downloaded from CHELSA. These then have to be processed locally to extract dynamic variables for each occurrence record co-ordinate and date. This is unlike GEE that remotely processed the dynamic variable data on Google servers. 

I could add a function to extract dynamic variables from downloaded CHELSA data, but this would have significantly higher computing storage and power demands than GEE extraction. 

Would this still be a desirable function to add to dynamicSDM?

**Reply by rogerio-bio** [reply]
Hello, 

Thank you for the reply.

Well, I think for testing, it would be a good option. Currently, **I use CHELSA and EarthEnv variables a lot**, many of which are not available in GEE. I know this is not the initial proposal for the package, but if it could be done, I would test it in an area of a conservation unit that I am currently working on.

**Reply by r-a-dobson** [reply]
Ok. I will write code for two functions: 
1) Extract dynamic variables from daily CHELSA data for each occurrence record given. 
2) Extract dynamic rasters from daily CHELSA data for projections. 

I will update here once they have been written. It would be great if you can test them for your study. Could you please give me an example variable that you would like to extract from daily CHELSA data?

**Reply by rogerio-bio** [reply]
Hello Dobson,

I could test bio04 (temperature seasonality), bio12 (annual precipitation amount) and bio15 (precipitation seasonality), or all of the climatologies data. I will start to clipping all of them for the conservation unit.

**Reply by r-a-dobson** [reply]
Sorry I must have misunderstood your request. As far as I am aware these datasets (BIO4,12 &15) are static. Please see the function extract_static_coords() and for SDM/SAM projections you can use the dynamic_proj_covariates() `static_rasters` argument and others that are associated.
